src/movies.js

```diff
@@ -5,31 +5,101 @@ console.log('movies: ', movies);
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
+function getAllDirectors(movies) {
+  return movies.map(function(movie) {
+    return movie.director;
+  })
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
+function howManyMovies(movies) {
+  return movies.filter(function(movie) {
+    return (movie.director === 'Steven Spielberg' && movie.genre.includes('Drama'));
+  })
+  .length;
+}
```

Comment:
Great syntax to return result straight away! Also return `.length `after filter method - excellent